xwiki-platform-core/xwiki-platform-web/xwiki-platform-web-war/src/main/webapp/resources/js/xwiki/editors/autosave.js

```diff
@@ -114,11 +112,11 @@ editors.AutoSave = Class.create({
     this.setTimeUnit();
     frequencyLabel.appendChild(document.createTextNode(" "));
     frequencyLabel.appendChild(this.timeUnit);
-    if (!this.options.enabled) {
-      frequencyLabel.setOpacity(this.options.disabledOpacity);
-    }
     // A paragraph containing the whole thing
     var container = new Element('div', {"id": "autosaveControl"});
+    if (!this.options.enabled) {
+      container.addClassName('disabled');
```

Comment:
I wonder if this class name isn't too generic and could conflict with other uses. For example, I found this (deprecated) style that adds an opacity on any element with class `disabled`: https://github.com/xwiki/xwiki-platform/blob/bc6fb452ef10414c1756b2b547e8129ec1e35c9d/xwiki-platform-core/xwiki-platform-web/xwiki-platform-web-war/src/main/webapp/resources/js/xwiki/editors/dataeditors.css#L190

I wonder if it wouldn't be enough to match on the checkbox state in CSS. The following selector seems to work in a simple test: `#autosaveControl:has(input:checked) label.frequency`.

Reply:
Just noticed that this doesn't work in Firefox as it doesn't support `:has`. :disappointed: So we need the class I fear. What about making the class more specific like `autosaveDisabled`?

Reply:
> Updated in the latest commit

Is it possible that forgot to push this commit?